huaweicloud/services/ddm/resource_huaweicloud_ddm_instance.go

```diff
@@ -0,0 +1,931 @@
+	// updateInstanceName: update DDM instance name
+	var (
+		updateInstanceNameHttpUrl = "v1/{project_id}/instances/{instance_id}/modify-name"
+		updateInstanceProduct     = "ddm"
+	)
+	updateInstanceNameClient, err := cfg.NewServiceClient(updateInstanceProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DDM client: %s", err)
+	}
+
+	updateInstanceNamePath := updateInstanceNameClient.Endpoint + updateInstanceNameHttpUrl
+	updateInstanceNamePath = strings.ReplaceAll(updateInstanceNamePath, "{project_id}", updateInstanceNameClient.ProjectID)
+	updateInstanceNamePath = strings.ReplaceAll(updateInstanceNamePath, "{instance_id}", fmt.Sprintf("%v", d.Id()))
+
+	updateInstanceNameOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+	}
+	updateInstanceNameOpt.JSONBody = utils.RemoveNil(buildUpdateInstanceNameBodyParams(d))
+	_, err = updateInstanceNameClient.Request("PUT", updateInstanceNamePath, &updateInstanceNameOpt)
+	if err != nil {
+		return diag.Errorf("error updating DdmInstance name: %s", err)
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{"MODIFY_NAME"},
+		Target:       []string{"RUNNING"},
+		Refresh:      ddmInstanceStatusRefreshFunc(d.Id(), region, cfg),
+		Timeout:      d.Timeout(schema.TimeoutUpdate),
+		PollInterval: 2 * time.Second,
+	}
+
+	_, err = stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return diag.Errorf("error waiting for instance (%s) to running: %s", d.Id(), err)
+	}
+	return nil
+}
+
+func updateInstanceSecurityGroup(_ context.Context, d *schema.ResourceData, cfg *config.Config, region string) diag.Diagnostics {
+	// updateInstanceSecurityGroup: update DDM instance security group
+	var (
+		updateInstanceSecurityGroupHttpUrl = "v1/{project_id}/instances/{instance_id}/modify-security-group"
+		updateInstanceProduct              = "ddm"
+	)
+	updateInstanceSecurityGroupClient, err := cfg.NewServiceClient(updateInstanceProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DDM client: %s", err)
+	}
+
+	updateInstanceSecurityGroupPath := updateInstanceSecurityGroupClient.Endpoint + updateInstanceSecurityGroupHttpUrl
+	updateInstanceSecurityGroupPath = strings.ReplaceAll(updateInstanceSecurityGroupPath, "{project_id}",
+		updateInstanceSecurityGroupClient.ProjectID)
+	updateInstanceSecurityGroupPath = strings.ReplaceAll(updateInstanceSecurityGroupPath, "{instance_id}",
+		fmt.Sprintf("%v", d.Id()))
+
+	updateInstanceSecurityGroupOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+	}
+	updateInstanceSecurityGroupOpt.JSONBody = utils.RemoveNil(buildUpdateInstanceSecurityGroupBodyParams(d))
+	_, err = updateInstanceSecurityGroupClient.Request("PUT", updateInstanceSecurityGroupPath, &updateInstanceSecurityGroupOpt)
+	if err != nil {
+		return diag.Errorf("error updating DdmInstance: %s", err)
```

Comment:
`return diag.Errorf("error updating DdmInstance security group: %s", err)`.
Try to make the error message clear. pls check below

Reply:
done